controllers/replication.storage/volumereplication_controller.go

```diff
@@ -363,16 +365,51 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	}
 
 	instance.Status.LastCompletionTime = getCurrentTime()
+
+	var requeueForInfo bool
+
+	if instance.Spec.ReplicationState == replicationv1alpha1.Primary {
+		info, err := r.getLastSyncTime(vr)
+		if err != nil {
+			logger.Error(err, "Failed to get volume replication info")
+			return ctrl.Result{}, err
+		}
+		lastSynctime := metav1.NewTime(info.AsTime())
+		instance.Status.LastSyncTime = &lastSynctime
+		requeueForInfo = true
+	}
 	err = r.updateReplicationStatus(instance, logger, getReplicationState(instance), msg)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
 	logger.Info(msg)
 
+	scheduleTime := getScheduleTime(parameters, logger)
+	if requeueForInfo {
+		logger.Info("volume is primary, requeuing to get volume replication info")
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: scheduleTime,
+		}, nil
+	}
+
 	return ctrl.Result{}, nil
 }
 
+func getScheduleTime(parameters map[string]string, logger logr.Logger) time.Duration {
```

Comment:
this can indeed be easily unit-tested, make sure to include the following cases:
- an empty map (no items)
- a map with a `schedulingInterval` item that is `nil`
- a map with a `schedulingInterval` item that is `""`
- a map with a `schedulingInterval` item that is can not be parsed by `time.ParseDuration()`
- a map with a `schedulingInterval` item that is valid

Reply:
this seems to be covered too